File: chapter1/009/main.py
def main():
	print(rotate("waterbottle", "erbottlewat"))

	print(rotate("stenio", "stenio"))
	print(rotate("stenioo", "ostenio"))
	print(rotate("stenioo", "enioest"))

def rotate(original, rotation):
    if len(original) == len(rotation):
        return original in (rotation + rotation)
    
    return False

main()

File: chapter1/009/main.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	fmt.Println(isRotation("waterbottle", "erbottlewat"))

	fmt.Println(isRotation("stenio", "stenio"))
	fmt.Println(isRotation("stenioo", "ostenio"))
	fmt.Println(isRotation("stenioo", "enioest"))
}

func isRotation(original, rotation string) bool {
	if len(original) != len(rotation) {
		return false
	}

	return strings.Contains(rotation+rotation, original)
}
